distributed/client.py

```diff
@@ -4089,18 +4094,21 @@ def register_worker_callbacks(self, setup=None):
         """
         return self.register_worker_plugin(_WorkerSetupPlugin(setup))
 
-    async def _register_worker_plugin(self, plugin=None, name=None):
-        responses = await self.scheduler.register_worker_plugin(
-            plugin=dumps(plugin, protocol=4), name=name
-        )
+    async def _register_worker_plugin(self, plugin=None, name=None, nanny=None):
+        if nanny or nanny is None and isinstance(plugin, NannyPlugin):
+            method = self.scheduler.register_nanny_plugin
+        else:
+            method = self.scheduler.register_worker_plugin
+
+        responses = await method(plugin=dumps(plugin, protocol=4), name=name)
         for response in responses.values():
             if response["status"] == "error":
                 exc = response["exception"]
                 tb = response["traceback"]
                 raise exc.with_traceback(tb)
         return responses
 
-    def register_worker_plugin(self, plugin=None, name=None, **kwargs):
+    def register_worker_plugin(self, plugin=None, name=None, nanny=None, **kwargs):
```

Comment:
Can you add `nanny=` to the list of parameters in this docstring?

Reply:
Also, is there a reason we want the default value to be `None` instead of `False`?

Reply:
Ah, after reading more I see we're doing a `nanny is None` check elsewhere. So please ignore my default value comment